MekHQ/src/mekhq/gui/utilities/StaticChecks.java

```diff
@@ -405,6 +405,42 @@ public static boolean doAllUnitsHaveC3Master(Vector<Unit> units) {
         return true;
 
     }
+    
+    /**
+     * Used to test a selection of Units provided by the player and determine whether they all share a designated unitType. 
+     * @param units Vector of units that the player has selected
+     * @return false if any unit in the passed-in Vector does not have the specified unit type
+     */
+    public static boolean areAllUnitsSameType(Vector<Unit> units, int unitType) {
+        // For our purposes, a selection of tanks isn't the same unless they're all the same weight class
+        // Set this based on the first unit in the selection
+        double firstUnitWeight = units.get(0).getEntity() != null ? units.get(0).getEntity().getWeight() : 0;
+        boolean light = firstUnitWeight > 0 && firstUnitWeight <= 50;
+        boolean heavy = firstUnitWeight > 50 && firstUnitWeight <= 100;
+        boolean superheavy = firstUnitWeight > 100 && firstUnitWeight <= 200;
```

Comment:
Is there something > 200 for units?

Reply:
There is not. I know you can construct such units, but the SH Vehicle *bay* only holds up to 200 tons - anything heavier is Cargo - so I think I'm ok here.

Reply:
There are a few examples, like the Koi and Ryu WiGE.

Reply:
But mkerensky is correct that vehicles over 200 tons cannot be transported in a bay.